cli/src/commands/download.ts

```diff
@@ -78,42 +81,73 @@ export default class Download extends Command {
       this.conf.set('projects', projects);
     }
     this.log('Downloading application...');
-    const targetDir = '.';
-    const extract = tar.extract({ cwd: targetDir });
+    const stagingDir = await mkdtemp(path.resolve(tmpdir(), 'reshuffle-download-'), { encoding: 'utf8' });
+    const extract = tar.extract({ cwd: stagingDir });
     const verboseLog = (type: string, err: Error) => {
       if (verbose) {
         this.log(`${type}:  ${err.message}`);
       }
     };
-    await new Promise<void>((resolve, reject) => {
-      fetch(downloadUrl)
-        .then((res) =>  {
-          const statusCode = res.status;
-          if (statusNotOk(statusCode)) {
-            reject(new CLIError(`Bad status code ${statusCode} when fetching ${downloadUrl}`));
-          }
-          res.body.pipe(extract)
-            .on('error', (err) => {
-              verboseLog('extract', err);
-              reject(new CLIError('Failed extracting application'));
-            })
-            .on('finish', () => {
-              resolve();
-            });
-        })
-        .catch((err) => {
-          verboseLog('download', err);
-          reject(new CLIError(`Failed fetching ${downloadUrl}`));
+    try {
+      await new Promise<void>((resolve, reject) => {
+        fetch(downloadUrl)
+          .then((res) =>  {
+            const statusCode = res.status;
+            if (statusNotOk(statusCode)) {
+              reject(new CLIError(`Bad status code ${statusCode} when fetching ${downloadUrl}`));
+            }
+            let first = true;
+            res.body.pipe(extract)
+              .on('error', (err) => {
+                if (first) {
+                  verboseLog('extract', err);
+                  first = false;
+                }
+                reject(new CLIError('Failed extracting application'));
+              })
+              .on('finish', () => {
```

Comment:
It seems like the correct event should be `close` in latest `tar` or `minipass` (see https://github.com/binaris/valet/pull/412).